x-pack/plugins/discover_enhanced/ui_tests/tests/saved_searches.spec.ts

```diff
@@ -0,0 +1,118 @@
+  expect(
+    await pageObjects.filterBar.hasFilter({
+      field: 'category',
+      value: `Men's Shoes`,
+    })
+  ).toBe(false);
+  await expect(page.testSubj.locator('queryInput')).toHaveText('');
+};
+
+test.describe(
+  'Discover app - saved searches',
+  { tag: ['@ess', '@svlSecurity', '@svlOblt', '@svlSearch'] },
+  () => {
+    const START_TIME = '2019-04-27T23:56:51.374Z';
+    const END_TIME = '2019-08-23T16:18:51.821Z';
+    const PANEL_NAME = 'Ecommerce Data';
+    const SEARCH_QUERY = 'customer_gender:MALE';
+    const SAVED_SEARCH_NAME = 'test-unselect-saved-search';
+
+    test.beforeAll(async ({ esArchiver, kbnClient, uiSettings }) => {
+      await esArchiver.loadIfNeeded(ES_ARCHIVES.ECOMMERCE);
+      await kbnClient.importExport.load(KBN_ARCHIVES.DISCOVER);
+      await kbnClient.importExport.load(KBN_ARCHIVES.ECOMMERCE);
+      await uiSettings.set({
+        defaultIndex: DATA_VIEW_ID.ECOMMERCE,
+        'doc_table:legacy': false,
+        'timepicker:timeDefaults': `{ "from": "${START_TIME}", "to": "${END_TIME}"}`,
+      });
+    });
+
+    test.afterAll(async ({ kbnClient, uiSettings }) => {
+      await uiSettings.unset('doc_table:legacy', 'defaultIndex', 'timepicker:timeDefaults');
+      await kbnClient.savedObjects.cleanStandardList();
+    });
+
+    test.beforeEach(async ({ browserAuth }) => {
+      await browserAuth.loginAsPrivilegedUser();
+    });
+
+    test('should customize time range on dashboards', async ({ pageObjects, page }) => {
+      await pageObjects.dashboard.goto();
+      await pageObjects.dashboard.openNewDashboard();
+      await pageObjects.dashboard.addPanelFromLibrary(PANEL_NAME);
+      await page.testSubj.locator('savedSearchTotalDocuments').waitFor({
+        state: 'visible',
+      });
+
+      await pageObjects.dashboard.customizePanel({
+        name: PANEL_NAME,
+        customTimeRageCommonlyUsed: { value: 'Last_90 days' },
+      });
+      await expect(
+        page.testSubj.locator('embeddedSavedSearchDocTable').locator('.euiDataGrid__noResults')
+      ).toBeVisible();
+    });
+
+    test(`should unselect saved search when navigating to a 'new'`, async ({
+      pageObjects,
+      page,
+    }) => {
+      await pageObjects.discover.goto();
+      await expect(page.testSubj.locator('*dataView-switch-link')).toHaveText(DATA_VIEW.ECOMMERCE);
+
+      await pageObjects.filterBar.addFilter({
+        field: 'category',
+        operator: 'is',
+        value: `Men's Shoes`,
+      });
+      await page.testSubj.fill('queryInput', SEARCH_QUERY);
+      await page.testSubj.click('querySubmitButton');
+      await pageObjects.discover.waitForHistogramRendered();
+
+      await pageObjects.discover.saveSearch(SAVED_SEARCH_NAME);
+      await pageObjects.discover.waitForHistogramRendered();
+
+      expect(
+        await pageObjects.filterBar.hasFilter({
+          field: 'category',
+          value: `Men's Shoes`,
+          enabled: true,
+        })
+      ).toBe(true);
```

Comment:
I'm wondering if we want to encourage usage of assertion messages, particularly for bool assertions. Right now, if this fails, it shows like this in the log:
```
  1) [chromium] › saved_searches.spec.ts:72:9 › Discover app - saved searches › should unselect saved search when navigating to a 'new' 

    Error: expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false
```
which doesn't tell us too much.  When using an assertion message like
```
      expect(
        await pageObjects.filterBar.hasFilter({
          field: 'category',
          value: `Men's Shoes`,
          enabled: true,
        }),
        `Filter {field: "category", value: "Men's Shoes", enabled: true} should exist`
      ).toBe(true);
```
the output would be better readable IMO:
```
  1) [chromium] › saved_searches.spec.ts:72:9 › Discover app - saved searches › should unselect saved search when navigating to a 'new' 

    Error: Filter {field: "category", value: "Men's Shoes", enabled: true} should exist

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false
```
What do you think?

Reply:
> I'm wondering if we want to encourage usage of assertion messages
We re-export `expect` from `kbn-scout`, so we can enforce requirement to always provide error message. 

I'm not sure if it is a good approach, but it definitely should bring a good practice in place. Your example is a proof. 
I didn't do it in this PR, we can discuss it in a group and apply later.